superduper/backends/base/data_backend.py

```diff
@@ -130,53 +108,154 @@ def drop(self, force: bool = False):
 
     @abstractmethod
-    def list_tables_or_collections(self):
-        """List all tables or collections in the database."""
+    def reconnect(self):
+        """Reconnect to the databackend."""
 
-    @staticmethod
-    def infer_schema(data: t.Mapping[str, t.Any], identifier: t.Optional[str] = None):
-        """Infer a schema from a given data object.
+    ########################################################
+    # Abstract methods/ optional methods to be implemented #
+    ########################################################
 
-        :param data: The data object
-        :param identifier: The identifier for the schema, if None, it will be generated
-        :return: The inferred schema
-        """
+    @abstractmethod
+    def insert(self, table: str, documents: t.Sequence[t.Dict]) -> t.List[str]:
+        pass
 
-    def check_ready_ids(
-        self, query: Query, keys: t.List[str], ids: t.Optional[t.List[t.Any]] = None
-    ):
-        """Check if all the keys are ready in the ids.
+    @abstractmethod
+    def missing_outputs(self, query: Query, predict_id: str) -> t.List[str]:
+        pass
 
-        :param query: The query object.
-        :param keys: The keys to check.
-        :param ids: The ids to check.
-        """
-        if ids:
-            query = query.select_using_ids(ids)
-        data = query.execute()
-        ready_ids = []
-        for select in data:
-            notfound = 0
-            for k in keys:
-                try:
-                    select[k]
-                except KeyError:
-                    notfound += 1
-            if notfound == 0:
-                ready_ids.append(select[query.primary_id])
-        self._log_check_ready_ids_message(ids, ready_ids)
-        return ready_ids
-
-    def _log_check_ready_ids_message(self, input_ids, ready_ids):
-        if input_ids and len(ready_ids) != len(input_ids):
-            not_ready_ids = set(input_ids) - set(ready_ids)
-            logging.info(f"IDs {not_ready_ids} do not ready.")
-            logging.debug(f"Ready IDs: {ready_ids}")
-            logging.debug(f"Not ready IDs: {not_ready_ids}")
+    @abstractmethod
+    def primary_id(self, query: Query) -> str:
+        pass
+
+    @abstractmethod
+    def select(self, query: Query) -> t.List[t.Dict]:
+        pass
+
+    def to_id(self, id: t.Any) -> str:
+        return id
+
+    ##########################################
+    # Methods which leverage implementations #
+    ##########################################
+
+    def get(self, query: Query):
+        assert query.type == 'select'
+
+        if query.decomposition.pre_like:
+            return list(self.pre_like(query, n=1))[0]
+
+        elif query.decomposition.post_like:
+            return list(self.post_like(query, n=1))[0]
+
+        return query.limit(1).execute()[0]
+
+    def _wrap_results(self, query: Query, result, schema):
+        pid = self.primary_id(query)
```

Comment:
Errors will occur for tables not created by SuperDuper. For example, tables that already exist in database cannot have their `schema` and `primary_id` retrieved.

Reply:
This isn't a changed behaviour I don't think.

Reply:
1. Insert data into a table using a non-superduper method (such as ibis, SQL queries, etc.)
2. Query the data from this table using superduper.

An error should occur.

Reply:
Resolved, since this is already an issue with current version. Potentially could add auto-schema also for select queries.